docs/devel/luns.md

```diff
@@ -0,0 +1,82 @@
+Direct LUNs migration
+==============
+
+Summary
+-------
+
+VM contains disks. Some of them might be connected as a direct LUN.
+When a user wish to re-use the LUN disks, he needs them attached to the VM.
+The LUN should be used in a single place, or else, it will cause a data corruption.
+The storage where the LUNs reside, needs to be reachable to the VM environment.
+This feature intend to support oVirt only.
+
+Design
+------
+
+The requirements for the LUN migration code are (not in priority order)
+
+1. Gather the LUN information from the source destination.
+2. To ease on the user - less manual actions, more automated migration.
+3. Avoid unnecessary risks.
+
+Requirements
+----------------------------------------
+
+1. Destination (OCP cluster) reachability to the ISCSI storage, where the LUN is.
+2. RBAC permission to create `PersistentVolumes` by the controller.
+   - Will be added to the operator manifest.
+
+Risks and Mitigations
+----------------------------------------
+
+1. A VM that will be used while or after the migration with a direct LUN connected.
+   - A warning should be shown before starting the migration.
+   - Detaching the LUN from source VM, once the migration completes.
+   - (Optional) UI checkbox to the user, before starting the migration.
+   - The migration _must_ be a cold migration.
+2. An oVirt version with unsupported API. The [PR](https://github.com/oVirt/ovirt-engine/pull/559) provides LUN details is required.
+   - We may check for missing LUN details and skip the disk migration.
+   - (Optional) Have a mechanism to check the oVirt version for future work.
+3. Creating a `PersistentVolume` requires the file system type: `ext4` or `xfs` of the LUN, which we don't know.
+   - Might not be required for `Block` type.
+4. On non-local destination, the controller needs permission to create the `PersistentVolumes`.
+   - It may be enough to be added to the forklift-operator.
```

Comment:
how would it work? 
maybe we just need to check for this permission and ask user to add it?

Reply:
TBH, i'm really not sure about it. adding it to the manifest in the operator should be enough when MTV is in the same cluster as the destination. the controller does have permissions to create CRs on destination cluster. If so, this is the place where we say what permission are granted to the controller. but only testing it will give the answer (or if anyone knows).